CMakeLists.txt

```diff
@@ -34,6 +34,33 @@ if (MSVC)
   set(CMAKE_CXX_STANDARD_REQUIRED ON)
 endif()
 
+set(COMPILER ${CMAKE_CXX_COMPILER_ID})
+if (COMPILER STREQUAL "GNU")
+  set(COMPILER "GCC") # perfer printing GCC instead of GNU
+endif()
+if (COMPILER MATCHES "Clang")
+  set(COMPILER "Clang") # Simplify all Clang variants to just Clang
```

Comment:
This wouldn't be a great idea, Apple Clang (what you get with XCode) can effectively be treated as a different compiler, as it lags behind standard Clang in features and doesn't align its versions with standard Clang.

Reply:
I thought they had reconverged. But looks like that was just a coincidence, and as of Xcode 14.3 beta, they are diverging again, this time in the opposite direction. 

Fine... easy enough to handle the separately.